src/color/readableColor.js

```diff
@@ -0,0 +1,46 @@
+// @flow
+
+import parseToRgb from './parseToRgb'
+import curry from '../internalHelpers/_curry'
+
+/**
+ * Selects black or white or best contrast depending on the luminosity of the supplied color.
+ * Follows W3C specs for readability
+ *
+ * @example
+ * // Styles as object usage
+ * const styles = {
+ *   color: mostReadable('#000'),
+ *   color: mostReadable('rgb(255,0,0)'),
+ * }
+ *
+ * // styled-components usage
+ * const div = styled.div`
+ *   color: ${mostReadable('#000')};
+ *   color: ${mostReadable('rgb(255,0,0)')};
+ * `
+ *
+ * // CSS in JS Output
+ *
+ * element {
+ *   color: "#fff";
+ *   color: "#000";
+ * }
+ */
+
+function readableColor(color: string): string {
+  const w3cConversion = (rgbColor: number): number => {
+    const colorZeroToOne = rgbColor / 255
+    return colorZeroToOne <= 0.03928
+      ? colorZeroToOne / 12.92
+      : ((colorZeroToOne + 0.055) / 1.055) ** 2.4
+  }
+  const parsedColor = parseToRgb(color)
+  const luminosity =
+    w3cConversion(parsedColor.red) * 0.2126 +
+    w3cConversion(parsedColor.green) * 0.7152 +
+    w3cConversion(parsedColor.blue) * 0.0722
+  return luminosity > 0.179 ? '#000' : '#fff'
+}
+
```

Comment:
Absolutely.  https://www.w3.org/TR/WCAG20-TECHS/G18.html

The only catch is that I've taken the liberty of doing some algebra that may not be apparent on the surface of that link.  I'll lay it out and see what you think.

The contrast equation is:  (L1 + 0.05) / (L2 + 0.05), where L1 is the luminance of lightest color and L2 the luminance of the darkest.  We know that the luminance of white is 1 and black is 0, so we can do some substitution... 

(L + 0.05) / (0.0 + 0.05) > (1.0 + 0.05) / (L + 0.05)  

If the left side is greater, use black, or else use white.  That equation reduces to:

if L > 0.179 use #000000 else use #ffffff

The rest of the constant are obvious by looking through the link.